onnxruntime/contrib_ops/cpu/activations.h

```diff
@@ -38,6 +39,34 @@ class Gelu : public OpKernel {
   Status Compute(OpKernelContext* context) const override {
     const auto* X = context->Input<Tensor>(0);
     Tensor* Y = context->Output(0, X->Shape());
+    concurrency::ThreadPool* tp = context->GetOperatorThreadPool();
+    if (nullptr != tp) {
+      const T* input = X->template Data<T>();
+      T* output = Y->template MutableData<T>();
+      int thread_num = tp->NumThreads();
```

Comment:
You may plus it to one.

Reply:
> You may plus it to one.

ugh. we really need to fix this. a user (internal or external) shouldn't need to know such things. NumThreads should return the number of threads the threadpool will use.

Reply:
Yes, we should fix this. But NumThreads() doesn't know if the caller is in the threadpool or not.

Reply:
The hole was digger by @yufenglee himself

Reply:
@snnn the real matter is whether we should use the main thread for calculation or not in ParallelFor. I remembered you've ever made a change to not use main thread for calculation and triggered a bad performance regression, and you reverted your change, i.e., doing calculation in main thread get better performance.

Reply:
Correct.